src/umdsst/Fields/Fields.cc

```diff
@@ -321,6 +321,52 @@ namespace umdsst {
     return atlasFieldSet_;
   }
 
+// ----------------------------------------------------------------------------
+
+  void Fields::setAtlas(atlas::FieldSet * fs) const {
+    fs->add((*atlasFieldSet_)[0]);
+  }
+
+// ----------------------------------------------------------------------------
+
+  void Fields::toAtlas(atlas::FieldSet * fs_to) const {
+    const int size = geom_->atlasFunctionSpace()->size();
+
+    // Ligang: you will have segment fault with following delete/new code.
+    // if (fs_to)
+    //   delete fs_to;
+    // fs_to = new atlas::FieldSet();
+
+    for (int i = 0; i < vars_.size(); i++) {
+      std::string var_name = vars_[i];
+      atlas::Field fld_to  = geom_->atlasFunctionSpace()->createField<double>(
```

Comment:
I think `toAtlas()` is supposed to check and only create a new field if it doesn't already exist. This function will be called many times in the var, and so the field only has to be created the first time, after that the contents are simply copied.

Reply:
Done.